Compile error on Mac M1

<!--
Thank you for taking the time to file an issue, we highly appreciate your contribution.

Before you go ahead, please make sure that the issue has not been reported already.
For support questions, please head over to our [Discord](http://near.chat/) or ask on [StackOverflow](https://stackoverflow.com/search?q=nearprotocol).
-->

**Describe the bug**
`cargo check --tests --benches` fails on top of master on Mac M1 machines.

**To Reproduce**
```
$ cargo check --tests --benches
    Checking near-vm-runner v0.0.0 (/Users/akhi/projects/nearcore/runtime/near-vm-runner)
    Checking nearcore v0.0.0 (/Users/akhi/projects/nearcore/nearcore)
    Checking near-jsonrpc-tests v0.0.0 (/Users/akhi/projects/nearcore/chain/jsonrpc/jsonrpc-tests)
error[E0432]: unresolved import `crate::wasmer2_runner`
  --> runtime/near-vm-runner/src/tests/fuzzers.rs:15:12
   |
15 | use crate::wasmer2_runner::Wasmer2VM;
   |            ^^^^^^^^^^^^^^ could not find `wasmer2_runner` in the crate root

error[E0432]: unresolved import `wasmer_engine`
  --> runtime/near-vm-runner/src/tests/fuzzers.rs:11:5
   |
11 | use wasmer_engine::Executable;
   |     ^^^^^^^^^^^^^ use of undeclared crate or module `wasmer_engine`

    Checking near-jsonrpc-fuzz v0.0.0 (/Users/akhi/projects/nearcore/chain/jsonrpc/fuzz)
For more information about this error, try `rustc --explain E0432`.
error: could not compile `near-vm-runner` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Expected behavior**
`cargo check --tests --benches` should succeed.

**Version (please complete the following information):**
- nearcore commit/branch: `72affb9`
- rust version: `rustc 1.63.0 (4b91a6ea7 2022-08-08)`
- local

--- a/runtime/near-vm-runner/src/tests/fuzzers.rs
+++ b/runtime/near-vm-runner/src/tests/fuzzers.rs
@@ -8,11 +8,9 @@ use near_primitives::version::PROTOCOL_VERSION;
 use near_vm_errors::{FunctionCallError, VMError};
 use near_vm_logic::mocks::mock_external::MockedExternal;
 use near_vm_logic::{VMConfig, VMContext};
-use wasmer_engine::Executable;
 
 use crate::internal::wasmparser::{Export, ExternalKind, Parser, Payload, TypeDef};
 use crate::internal::VMKind;
-use crate::wasmer2_runner::Wasmer2VM;
 use crate::VMResult;
 
 /// Finds a no-parameter exported function, something like `(func (export "entry-point"))`.
@@ -171,8 +169,12 @@ fn wasmer2_and_wasmtime_agree() {
     });
 }
 
+#[cfg(all(feature = "wasmer2_vm", target_arch = "x86_64"))]
 #[test]
 fn wasmer2_is_reproducible() {
+    use crate::wasmer2_runner::Wasmer2VM;
+    use wasmer_engine::Executable;
+
     bolero::check!().for_each(|data: &[u8]| {
         if let Ok(module) = ArbitraryModule::arbitrary(&mut arbitrary::Unstructured::new(data)) {
             let code = ContractCode::new(module.0.module.to_bytes(), None);

